testnet/stacks-node/src/nakamoto_node/relayer.rs

```diff
@@ -172,6 +189,77 @@ impl LastCommit {
     }
 }
 
+pub type MinerThreadJoinHandle = JoinHandle<Result<(), NakamotoNodeError>>;
+
+/// Miner thread join handle, as well as an "abort" flag to force the miner thread to exit when it
+/// is blocked.
+pub struct MinerStopHandle {
+    /// The join handle itself
+    join_handle: MinerThreadJoinHandle,
+    /// The relayer-set abort flag
+    abort_flag: Arc<AtomicBool>,
+}
+
+impl MinerStopHandle {
+    pub fn new(join_handle: MinerThreadJoinHandle, abort_flag: Arc<AtomicBool>) -> Self {
+        Self {
+            join_handle,
+            abort_flag,
+        }
+    }
+
+    /// Get a ref to the inner thread object
+    pub fn inner_thread(&self) -> &std::thread::Thread {
+        self.join_handle.thread()
+    }
+
+    /// Destroy this stop handle to get the thread join handle
+    pub fn into_inner(self) -> MinerThreadJoinHandle {
+        self.join_handle
+    }
+
+    /// Set the miner-abort flag to true, which causes the miner thread to exit if it is blocked.
+    pub fn set_abort_flag(&self) {
+        self.abort_flag.store(true, Ordering::SeqCst);
+    }
+
+    /// Get an Arc to the abort flag, so another thread can set it.
+    pub fn get_abort_flag(&self) -> Arc<AtomicBool> {
+        self.abort_flag.clone()
+    }
```

Comment:
Becaause this struct is only used in this file, I think its simpler to remove these getter/setters and just use the instance variables directly.

Reply:
`.inner_thread()` and `.stop()` are used in `miner.rs` as well.